File: src/main/java/cube8540/oauth/authentication/credentials/oauth/code/domain/AuthorizationCodeRepository.java
package cube8540.oauth.authentication.credentials.oauth.code.domain;

import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AuthorizationCodeRepository extends JpaRepository<OAuth2AuthorizationCode, AuthorizationCode> {

    @Override
    @EntityGraph(attributePaths = {"approvedScopes"}, type = EntityGraph.EntityGraphType.LOAD)
    Optional<OAuth2AuthorizationCode> findById(AuthorizationCode code);
}

File: src/main/kotlin/cube8540/oauth/authentication/oauth/token/domain/AuthorizationCodeRepository.kt
package cube8540.oauth.authentication.oauth.token.domain

import java.util.Optional
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface AuthorizationCodeRepository: JpaRepository<OAuth2AuthorizationCode, String> {

    @EntityGraph(attributePaths = ["approvedScopes"], type = EntityGraph.EntityGraphType.LOAD)
    override fun findById(id: String): Optional<OAuth2AuthorizationCode>
}